allennlp/models/encoder_decoders/wikitables_semantic_parser.py

```diff
@@ -271,6 +310,166 @@ def forward(self,  # type: ignore
             # TODO(matt): compute accuracy here.
             return outputs
 
+    @staticmethod
+    def _get_neighbor_indexes(worlds: List[WikiTablesWorld],
+                              num_entities: int,
+                              tensor: torch.Tensor) -> Variable:
+        """
+        This method returns the indices of each entity's neighbors. A tensor
+        is accepted as a parameter for copying purposes
+
+        Parameters
+        ----------
+        worlds : ``List[WikiTablesWorld]``
+        num_entities : ``int``
+        tensor : ``torch.Tensor``
+            Used for copying the constructed list onto the right device.
+
+        Returns
+        -------
+        A ``torch.autograd.Variable`` with shape ``(batch_size, num_entities, num_entities)``.
+        Dimension 2 contains the indices for the neighbors of the entity represented by the
+        index of dimension 1. Dimension 2 is limited to num_entities as it is assumed that
+        each entity can have a number of neighbors at most equal to the total number of entities.
+        """
+        batches_neighbors = []
+        for world in worlds:
+            # Each batch instance has its own world, which has a corresponding table.
+            entities = world.table_graph.entities
+            entity2indices = {entity: i for i, entity in enumerate(entities)}
+            entity2neighbors = world.table_graph.neighbors
+            neighbor_indexes = []
+            for entity in entities:
+                entity_neighbors = [entity2indices[cn] for cn in entity2neighbors[entity]]
```

Comment:
What is `cn`?  `current_neighbor`?  A simple `n` would probably be better here, because it's more obviously just a temporary variable in a list comprehension, and it makes me less likely to try to think about what it's supposed to mean.